### DISCUSSION TITLE
Feature request: Action Buttons Similar to Add To Cart

### DISCUSSION THREAD

**Discussion by rpettique** [algolia/instantsearch#5610] [Ideas]
I tried searching for this feature but could not find anything. Is it possible to add action buttons to the results? Looks like there is something similar to what we are looking for in Add To Cart. Our use case would be if we were showing results for documentation (PDF's for example), that we could show a download icon in the results so they could immediately download the file without having to visit it's corresponding page. Also, possibly adding a share button. Any type of actionable button that would provide a shortcut from having to visit the contents page.

**Comment by Haroenv** [comment]
You can put any html you want, as well as click handlers in the template using the `html` tagged template:

```js
hits({
  // ...
  templates: {
    item(hit, { html, components }) {
      return html`<div>
        ${components.Highlight({ hit, attribute: 'name' })}
        <button onClick=${() => console.log('clicked', hit)}>
          do something
        </button>
      </div>`;
    },
  },
});
```
